src/System.IO.Pipelines/System/Buffers/MemoryPoolSlab.cs

```diff
@@ -9,7 +9,7 @@ namespace System.Buffers
     /// Slab tracking object used by the byte buffer memory pool. A slab is a large allocation which is divided into smaller blocks. The
     /// individual blocks are then treated as independant array segments.
     /// </summary>
-    public class MemoryPoolSlab : IDisposable
+    internal class MemoryPoolSlab : OwnedMemory<byte>
```

Comment:
Why is this an OwnedMemory?

Reply:
The allocation/deallocation callbacks need OwnedMemory. Since you call the callbacks when slabs are allocated/deallocated, I either needed to inherit slab from OwnedMemory or create a temp (throwaway) OwnedMemory when calling the callbacks. 

I also considered the callbacks to take something else than OwnedMemory (e.g. tuple of pointer/len), but I think the pool creates an illusion of OwnedMemory instances being the unit of allocation, and so OwnedMemory as a parameter of the callbacks seems to be consistent with the abstraction.

Reply:
Are callbacks only used in RIO sample, @davidfowl?

Reply:
I'd rather remove the callbacks altogether and keep the slabs as is. Copy this pool as is into the RIO code and make this one internal.

Reply:
@pakrym yes it is.